single-line-diagram/single-line-diagram-core/src/main/java/com/powsybl/sld/svg/DefaultLabelProvider.java

```diff
@@ -168,18 +168,24 @@ private <T extends Identifiable<T>> void addOperatingStatusDecorator(List<NodeDe
             OperatingStatus<T> operatingStatus = identifiable.getExtension(OperatingStatus.class);
             if (operatingStatus != null) {
                 switch (operatingStatus.getStatus()) {
-                    case PLANNED_OUTAGE -> nodeDecorators.add(getBranchStatusDecorator(node, direction, PLANNED_OUTAGE_BRANCH_NODE_DECORATOR));
-                    case FORCED_OUTAGE -> nodeDecorators.add(getBranchStatusDecorator(node, direction, FORCED_OUTAGE_BRANCH_NODE_DECORATOR));
+                    case PLANNED_OUTAGE -> nodeDecorators.add(getOperatingStatusDecorator(node, direction, PLANNED_OUTAGE_BRANCH_NODE_DECORATOR));
+                    case FORCED_OUTAGE -> nodeDecorators.add(getOperatingStatusDecorator(node, direction, FORCED_OUTAGE_BRANCH_NODE_DECORATOR));
                     case IN_OPERATION -> { /* No decorator for IN_OPERATION equipment */ }
                 }
             }
         }
     }
 
-    private NodeDecorator getBranchStatusDecorator(Node node, Direction direction, String decoratorType) {
-        return (node instanceof Middle3WTNode middle3WTNode) ?
-                new NodeDecorator(decoratorType, getMiddle3WTDecoratorPosition(middle3WTNode, direction)) :
-                new NodeDecorator(decoratorType, getFeederDecoratorPosition(direction, decoratorType));
+    private NodeDecorator getOperatingStatusDecorator(Node node, Direction direction, String decoratorType) {
+        if (node instanceof Middle3WTNode middle3WTNode) {
+            return new NodeDecorator(decoratorType, getMiddle3WTDecoratorPosition(middle3WTNode, direction));
+        } else if (node instanceof BusNode) {
+            return new NodeDecorator(decoratorType, getBusDecoratorPosition());
+        } else if (node instanceof FeederNode) {
+            return new NodeDecorator(decoratorType, getFeederDecoratorPosition(direction, decoratorType));
+        } else {
+            return new NodeDecorator(decoratorType, getInternal2WTDecoratorPosition(node.getOrientation()));
```

Comment:
As you're calling something which is called `getInternal2wtXxxx`, you'd better check beforehand if it's a `Internal2WTNode` instance.
This will add an else in which you would call `getGenericDecoratorPosition` (no usages so far), or otherwise you should delete this unused method.